File: server/Domain/Room.cs
namespace server.Models.Domain
{
    public class Room
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public int? OfficeId { get; set; } 
        public Office Office { get; set; } = null!;
        public ICollection<Seat> Seats { get; set; } = new List<Seat>();
    }
}

File: client/src/domain/room.js
import Seat from "./seat";

export default class Room {
  constructor(id = null, name = null, seats = []) {
      this.id = id;
      this.name = name;
      this.seats = seats.map(seat => new Seat(seat.id, seat.roomId, seat.isAvailable, seat.bookings));
  }
}